### PR TITLE
[FE] "Combine columns" drill thru

### LINKED ISSUE #39980: [FE] "Combine columns" drill thru
Depends on #39978.

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #39980

### Description
Adds combine column drill through to the chill mode header columns.

### How to verify

- Open the `Accounts` table in chill mode
- Click the header of a string column (eg. `Email`, `First Name` or `Plan`)
- Click `+ Combine columns`
- Play around with the column editor
- Click `Done`
- Verify the new column is added as the last column

### Demo

See video attached to [this Slack message](https://metaboat.slack.com/archives/C0645JP1W81/p1710427151050319)

Screenshot 2024-04-19 at 14 52 22
Screenshot 2024-04-19 at 14 52 26
Screenshot 2024-04-19 at 14 52 34

### CODE REVIEW

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.tsx
Comment (kamilmielnik) [inline frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.tsx:57]: We have `marginTop` defined for most Mantine inputs by default.
I think it's because of the assumption that if the input is `:not(:only-child)` there will be a label above it.
This assumption does not hold for `Select` because it never is the only child due to `<input type="hidden"` being rendered before it (see [screenshot](https://github.com/metabase/metabase/assets/6830683/ba484c9a-99fe-400b-9823-d20984b195fd)).

It'd be better to assume that every label that is `:not(:only-child)` will have an input underneath, and add `marginBottom` to labels instead. Though it'd be ideal if the low-level building blocks never made such assumptions at all.

Anyway, fixing that is out of the scope of this PR, so I added this band-aid here.

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.module.css
Comment (kamilmielnik) [inline frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.module.css:1]: It's not clear to me which styling should go to CSS modules, and which should be overridden with `styles` prop in Mantine components. Suggestions are appreciated.

File: frontend/src/metabase-lib/metadata.ts
Comment (kamilmielnik) [inline frontend/src/metabase-lib/metadata.ts:170]: Mocks compensating for the lack of BE implementation.
Don't bother reviewing it - this file should be reverted before the PR is merged.

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx
Comment (kamilmielnik) [inline frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx:111]: Mantine does not correctly handle it in RTL languages, i.e. the icon will be on the left in RTL layout as well.

File: frontend/src/metabase-lib/expression.ts
Comment (kamilmielnik) [inline frontend/src/metabase-lib/expression.ts:113]: Mocks compensating for the lack of BE implementation.

**Comment by kamilmielnik** [comment]
Dismissing request for review for now due to uncertainty on how we want to shape the UI here.

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.module.css
Comment (kamilmielnik) [inline frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.module.css:16]: We have `marginTop` defined for most Mantine inputs by default.
I think it's because of the assumption that if the input is `:not(:only-child)` there will be a label above it.
This assumption does not hold for `Select` because it never is the only child due to `<input type="hidden"` being rendered before it (see [screenshot](https://github.com/metabase/metabase/assets/6830683/ba484c9a-99fe-400b-9823-d20984b195fd)).

It'd be better to assume that every label that is `:not(:only-child)` will have an input underneath, and add `marginBottom` to labels instead. Though it'd be ideal if the low-level building blocks never made such assumptions at all.

Anyway, fixing that is out of the scope of this PR, so I added this band-aid here.

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/utils.ts
Comment (romeovs) [inline frontend/src/metabase/querying/utils/drills/combine-columns-drill/utils.ts:55]: This will cause `formatSeparator` to return `(empty)` in most cases. 

You probably want:
```suggestion
  if (!separator) {
```

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx
Comment (romeovs) [inline frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx:177]: We should probably disable the button when the form is in an invalid state.

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/utils.ts
Comment (kamilmielnik) [reply frontend/src/metabase/querying/utils/drills/combine-columns-drill/utils.ts]: Fixed on Friday, forgot to push: 9bd5d51e3b6f0e4ed26dd5467a306da3984faa2f :)

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx:177]: I don't think this form can be in an invalid state. Am I missing something?

Comment (romeovs) [reply frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx:177]: I guess it's a decision we can make, but when a column dropdown I think the empty the form is in an invalid state. The alternative is that we just ignore that column in the resulting expression.

Comment (kamilmielnik) [reply frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/CombineColumnsDrill/CombineColumnsDrill.tsx:177]: In this branch column dropdown is never empty.
When adding a new "row", a column is automatically selected, see `getNextColumnAndSeparator` and how it's used.

File: frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.module.css
Comment (romeovs) [reply frontend/src/metabase/querying/utils/drills/combine-columns-drill/components/ColumnAndSeparatorRow/ColumnAndSeparatorRow.module.css]: Fixed here: https://github.com/metabase/metabase/pull/41728

**Review by nemanjaglumac — COMMENTED** [review]
I'm seeing some weirdness with the column position when switching between the visualization and the raw table views.
https://www.loom.com/share/a7cfa7b8b09d4314955ec03dfcfdd786?sid=071f3a6b-86ea-46e7-9cfa-82ffe5906cea

**Comment by romeovs** [comment]
> I'm seeing some weirdness with the column position when switching between the visualization and the raw table views. https://www.loom.com/share/a7cfa7b8b09d4314955ec03dfcfdd786?sid=071f3a6b-86ea-46e7-9cfa-82ffe5906cea

Is that not was is expected?

The way I understand it, the table view is showing the "raw" table and thus showing the column at the end, whereas the visualization as a table is showing the "customized" table.

You can do the same with any column (even non-generated ones) in the table and it would yield the same issue.

**Review by nemanjaglumac — APPROVED** [review]

**Pull request merged**
